src/jdk.jdwp.agent/unix/native/libjdwp/exec_md.c

```diff
@@ -88,6 +88,9 @@ dbgsysExec(char *cmdLine)
         argv[i] = p;
         p = skipNonWhitespace(p);
         *p++ = '\0';
+        if ((i + 1) == argc) {
+            break; // finished: do not call skipWhitespace again
+        }
```

Comment:
It seems a bit odd, because you are breaking out of the middle of the loop by using a breakout condition that is already in the `for` statement, and now serves no purpose. I think I like the idea of moving `skipWhitespace(p)` to the start of the loop. No need to skip in on the first iteration. It will just return p.

Reply:
Yes, could have removed the condition from the for loop, but doing that can look surprising/unusual, so I left it in.

Putting skipWhitespace at the start is maybe more elegant? 8-)  No extra arithmetic to read.